pallets/encrypted-transactions/src/mock.rs

```diff
@@ -0,0 +1,194 @@
+    type Event = TestEvent;
+    type BlockHashCount = BlockHashCount;
+    type MaximumBlockWeight = MaximumBlockWeight;
+    type DbWeight = ();
+    type BlockExecutionWeight = ();
+    type ExtrinsicBaseWeight = ();
+    type MaximumExtrinsicWeight = MaximumBlockWeight;
+    type MaximumBlockLength = MaximumBlockLength;
+    type AvailableBlockRatio = AvailableBlockRatio;
+    type Version = ();
+    type PalletInfo = ();
+    type AccountData = ();
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+    type SystemWeightInfo = ();
+}
+
+impl orml_tokens::Trait for Test {
+    type Event = TestEvent;
+    type Balance = Balance;
+    type Amount = Amount;
+    type CurrencyId = TokenId;
+    type OnReceived = ();
+    type WeightInfo = ();
+}
+
+parameter_types! {
+    pub const NativeCurrencyId: u32 = NATIVE_CURRENCY_ID;
+    pub const TreasuryModuleId: sp_runtime::ModuleId = sp_runtime::ModuleId(*b"py/trsry");
+    pub const BnbTreasurySubAccDerive: [u8; 4] = *b"bnbt";
+}
+
+mod currency {
+    use mangata_primitives::Balance;
+
+    pub const MILLICENTS: Balance = 1_000_000_000;
+    pub const CENTS: Balance = 1_000 * MILLICENTS; // assume this is worth about a cent.
+    pub const DOLLARS: Balance = 100 * CENTS;
+}
+
+parameter_types! {
+    pub const EncryptedTxnsFee: Balance = 1 * currency::DOLLARS;
+    pub const DoublyEncryptedCallMaxLength: u32 = 4096;
+}
+
+use pallet_session::{SessionHandler, SessionManager};
+use sp_staking::SessionIndex;
+
+pub struct TestSessionManager;
+impl SessionManager<u64> for TestSessionManager {
+    fn end_session(_: SessionIndex) {}
+    fn start_session(_: SessionIndex) {}
+    fn new_session(_: SessionIndex) -> Option<Vec<u64>> {
+        None
+    }
+}
+
+use sp_runtime::testing::UintAuthorityId;
+use sp_runtime::traits::OpaqueKeys;
+
+pub struct TestSessionHandler;
+use sp_runtime::RuntimeAppPublic;
+
+impl SessionHandler<u128> for TestSessionHandler {
+    const KEY_TYPE_IDS: &'static [sp_runtime::KeyTypeId] = &[UintAuthorityId::ID];
+    fn on_genesis_session<T: OpaqueKeys>(_validators: &[(u128, T)]) {}
+    fn on_new_session<T: OpaqueKeys>(
+        _changed: bool,
+        _validators: &[(u128, T)],
+        _queued_validators: &[(u128, T)],
+    ) {
+    }
+    fn on_disabled(_validator_index: usize) {}
+    fn on_before_session_ending() {}
+}
+
+sp_runtime::impl_opaque_keys! {
+    pub struct MockSessionKeys {
+        pub dummy: UintAuthorityId,
+    }
+}
+
+impl From<UintAuthorityId> for MockSessionKeys {
+    fn from(dummy: UintAuthorityId) -> Self {
+        Self { dummy }
+    }
+}
+
+impl pallet_session::Trait for Test {
+    type Event = TestEvent;
+    type ValidatorId = <Self as frame_system::Trait>::AccountId;
+    type ValidatorIdOf = sp_runtime::traits::ConvertInto;
+    type ShouldEndSession = pallet_session::PeriodicSessions<(), ()>;
+    type NextSessionRotation = ();
+    type SessionManager = ();
+    type SessionHandler = TestSessionHandler;
```

Comment:
This would push new keys for type `dummy: UintAuthorityId`. But later on we are using `crate::ecdsa::AuthorityId` for the pallet config.

1. How is this working?
2. Why not just work with `crate::ecdsa::AuthorityId` or `dummy: UintAuthorityId` all the way?